sdk/eventhub/Azure.Messaging.EventHubs/assets/live-tests-azure-setup.ps1

```diff
@@ -271,35 +193,17 @@ try
 {
     # Create the service principal and grant contributor access for management in the resource group.
 
-    Write-Host "`t...Creating new service principal"
     Start-Sleep 1
 
-    $credentials = New-Object Microsoft.Azure.Commands.ActiveDirectory.PSADPasswordCredential -Property @{StartDate=Get-Date; EndDate=Get-Date -Year 2099; Password="$(GenerateRandomPassword)"}            
-    $principal = (New-AzADServicePrincipal -DisplayName "$($ServicePrincipalName)" -PasswordCredential $credentials)
+    $credentials = New-Object Microsoft.Azure.Commands.ActiveDirectory.PSADPasswordCredential -Property @{StartDate=Get-Date; EndDate=Get-Date -Year 2099; Password="$(GenerateRandomPassword)"}                 
+
+    $principal = (CreateServicePrincipal -Role "Contributor" -ServicePrincipalName "$($ServicePrincipalName)" -Credentials $Credentials -ResourceGroupName "$($ResourceGroupName)")

```

Comment:
Definitely need to test to see if `Contributor` allows data ownership as well.   From my understanding of the Azure RBAC tweaks, I don't believe that it does, but am not sure.

Reply:
I will come back to you on this

Reply:
As you said, the ```contributor``` role does not allow to send events.

Reply:
Damn.  I was hoping my information was out of date.